### PR TITLE
feat: automatically merge Dependabot PRs on approval

### LINKED ISSUE #373: Auto-merge Dependabot PRs on approval.
I think it would make our lives a little easier to automatically merge Dependabot PRs upon approval.

Unfortunately, [Dependabot’s configuration](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file) doesn’t support such a feature so we may want to consider the [`pull_request_review` event with `submitted` type](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved). There’s some more useful information in the Github’s [Enable auto-merge on a pull request](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request) as well.

### PR DESCRIPTION
**Pull request by jenstroeger**
Closes #373

### CODE REVIEW

File: .github/workflows/dependabot-automerge.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+on:
+  pull_request_review:
+    types: [submitted]
+
+permissions:
+  pull-requests: write
+
+jobs:
+  comment:
+    if: ${{ github.event.review.state == 'approved' && github.event.pull_request.user.login == 'dependabot[bot]' }}
```
Comment (behnazh) [inline .github/workflows/dependabot-automerge.yaml:14]: Is it possible to set the git user to `dependabot[bot]` and bypass this check in a PR?

Comment (jenstroeger) [reply .github/workflows/dependabot-automerge.yaml:14]: What do you mean?

I wanted this workflow to run for PRs opened _only_ by Dependabot, but I couldn’t find a better way to check that than
```yaml
github.event.pull_request.user.login == 'dependabot[bot]'
```

File: .github/workflows/dependabot-automerge.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Merge Dependabot PR
+      run: gh pr comment --body "@dependabot squash and merge" "$PR_URL"
+      env:
+        PR_URL: ${{ github.event.pull_request.html_url }}
+        GITHUB_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
```
Comment (behnazh) [inline .github/workflows/dependabot-automerge.yaml:21]: Let's create a new token with permissions required for a pull request check (untrusted context).

File: .github/workflows/dependabot-automerge.yaml (hunk 1 above)
Comment (behnazh) [reply .github/workflows/dependabot-automerge.yaml:14]: I was wondering if it's possible to impersonate `dependabot[bot]`. If it's not possible, we can assume this check works.

**Review by behnazh — DISMISSED** [review]

File: .github/workflows/dependabot-automerge.yaml (hunk 1 above)
Comment (jenstroeger) [reply .github/workflows/dependabot-automerge.yaml:14]: We tried it, can’t be impersonated by changing my local `git.name` settings.

File: .github/workflows/dependabot-automerge.yaml (hunk 2 above)
Comment (jenstroeger) [reply .github/workflows/dependabot-automerge.yaml]: Commit https://github.com/jenstroeger/python-package-template/pull/390/commits/996c5a5399a52c2359e631085a9af9baada0b4de.

**Review by behnazh — APPROVED** [review]

File: .github/workflows/dependabot-automerge.yaml (hunk 2 above)
Comment (jenstroeger) [reply .github/workflows/dependabot-automerge.yaml]: Merged.

**Pull request merged**